src/core/mapping/store.h

```diff
@@ -0,0 +1,215 @@
+/* Copyright 2022 NVIDIA Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include "core/data/transform.h"
+#include "core/utilities/typedefs.h"
+
+namespace legate {
+namespace mapping {
+
+class RegionField {
```

Comment:
Why do we have two `Store` and two `RegionField` classes now? one here and one in src/core/data?

Reply:
The two serve different roles; those in the `legate` namespace are handles for physical data access, whereas those in the `legate::mapping` namespace only hold the metadata for mapping.